src/daemon/daemonlocalserverconnection.cpp

```diff
@@ -160,3 +171,60 @@ void DaemonLocalServerConnection::write(const QJsonObject& obj) {
   m_socket->write(QJsonDocument(obj).toJson(QJsonDocument::Compact));
   m_socket->write("\n");
 }
+
+bool DaemonLocalServerConnection::isCallerAuthorized(const QString& command) {
+#if defined(MZ_MACOS)
+  uid_t uid;
+  gid_t gid;
+  int result = getpeereid(m_socket->socketDescriptor(), &uid, &gid);
+  if (result == -1) {
+    logger.error() << "Unable to determine if caller is authorized."
+                      "Assuming unauthorized.";
+    return false;
+  }
+
+  if (uid == 0) {
+    logger.debug() << "Command issued by root.";
+    // Case: This is root calling, root is always authorized.
+    return true;
+  }
+
+  QJsonObject status = Daemon::instance()->getStatus();
+  bool isConnected = status.value("connected").toBool();
+
+  if (isConnected) {
+    if (m_sessionUid == uid) {
+      // Case: VPN is currently active and caller is the one who activated it.
+      // Caller can access the APIs.
+      return true;
+    }
+
+    // Case: VPN is currently active and caller is **not** one who activated it.
+    // Caller **can't** access the APIs.
+    logger.error() << "Command issued by unexpected user" << uid;
+    return false;
+  }
+
+  // Case: VPN is not connected and we got a request to check the status, fine.
+  if (command == "status") {
+    return true;
+  }
```

Comment:
Allowing this might not actually be necessary if we change how things are done on the local socket controller. As things stand right now the local socket controller has a bunch of checkStatus calls while the VPN is not connected. Allowing this feels harmless, so I left it like this... 🤷‍♀️

Reply:
I think this is fine, checking the status is idempotent and it doesn't divulge any information that a user wouldn't be able to gather from inspecting the system anyways.